hw/lib/signup/sign_up_screen.dart

```diff
@@ -0,0 +1,313 @@
+                            ),
+                          ],
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Assets.female.image(width: 89, height: 88),
+                        ),
+                        Text( Batch16String.current.createAccount, style: TextStyle(fontWeight: FontWeight.w400, fontSize: 12),),
+                        Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+
+                            BlocListener<SignUpCubit, SignUpState>(
+                              listener: (context, state) {
+                                if (state is SignUpFailed  && state.provider =='github') {
+                                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(Batch16String.current.notSupported('Github'))));
+                                }
+                              },
+                              child: AccountHolder(image: Assets.git.image(width:  43, height:  43),
+                                  onTap: () {
+                                    context.read <SignUpCubit>().signupWithGitHub();
+
+
+                                  } ),
+                            ),
+                            AccountHolder(image: Assets.google.image(width:  43, height:  43),
+                              onTap: () =>
+                                  context.read<SignUpCubit>().signupWithGoogle()
+
+                              ,),
+                            AccountHolder(image: Assets.linkedIN.image(width:  43, height:  43),
+                                onTap: () =>
+                                    context.read<SignUpCubit>().signupWithLinkedIn())
+
+                          ],
+                        ),
+                        BlocSelector<SignUpCubit, SignUpState, String?>(
+                          selector: (state) {
+                            if (state is SignUpFailed)   {
+                              return state.provider;
+                            }
+                            return null;
+
+                          },
+                          builder: (context, state) {
+                            if (state == 'google') {
+                              return Text (Batch16String.current.notSupported("$state"));
+                            }
+                            return SizedBox.shrink();
+                          },
+
+                        ),
+                        Text('or', style: TextStyle(fontWeight: FontWeight.w400, fontSize: 13),),
+                        SizedBox(height: 20),
+
+                        IconTextFieldRow(hint: 'Name', image: Assets.human.path,
+                          onTextChange: (value) {
+                            currName = value;
```

Comment:
please set the name/email/password as value inside cubit. Try to avoid keeping params in view

Reply:
with your current implementation, we can keep param inside cubit, and update it when text name

```
class SignUpCubit extends Cubit<SignUpState> {
  SignUpCubit() : super(SignUpInitial());

  String _name;
  String _email;
  String _password;

  set name(String name) {
    _name = name;
  }

  set email(String email) {
    _email = email;
  }

  set password(String password) {
    _password = password;
  }
}
```

then in you `onTextChange`, you can write this 
```
    context.read<SignUpCubit>().name = value;
```

and finally, in SignUpCubit, create 1 sign up with email method
```
void signUpWithEmail() {
  if (_name.isEmpty) {
    emit(SignUpFailed('email', 'name'));
  }

  // other logic for other fields here
}
```